### PR TITLE
Rst docs

### PR DESCRIPTION
**Pull request by miklosduma**
The main change is `usage.rst` that doubles as a tutorial and also as a test suite.

It is only supported in Python 3.+

### CODE REVIEW

File: docs/make.bat [hunk 1]
```diff
@@ -0,0 +1,35 @@
+@ECHO OFF
```
Comment (lucafavatellawork) [inline docs/make.bat]: This seems a Windows script. Is this script invoked in any CI? Can it be deleted?

File: Makefile [hunk 2]
```diff
@@ -108,6 +110,7 @@ default-test:
 		&& $(PYTHON) -m sparkl_cli connect "$(TEST_SSE)" \
 		&& $(PYTHON) -m sparkl_cli login -r "$(TEST_USER)" "$(TEST_PASS)" \
 		&& curl -o "$(TEST_CERT)" "$(TEST_SSE)"/sse_cfg/claim -d email="$(TEST_ADMIN_USER)" -d password="$(TEST_ADMIN_PASS)" \
+		&& $(PYTHON) -m sparkl_cli close \
```
Comment (lucafavatellawork) [inline Makefile]: 👌
Do you recall what the symptom was? Spurious connection presented at beginning of tests?

File: docs/make.bat (hunk 1 above)
Comment (miklosduma) [reply docs/make.bat]: These files are used by Sphinx to auto-generate the docs. Sphinx supports both Windows and Unix/Linux. Right now there's not much to generate so I'm open to removing some/all of the Sphinx-generated stuff except for `usage.rst`

File: Makefile (hunk 2 above)
Comment (miklosduma) [reply Makefile]: The rst tests also use the default alias, hence I must close it before running them

File: docs/Makefile [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+# Put it first so that "make" without argument is like "make help".
+help:
+	@$(SPHINXBUILD) -M help "$(SOURCEDIR)" "$(BUILDDIR)" $(SPHINXOPTS) $(O)
+
+.PHONY: help Makefile
+
+# Catch-all target: route all unknown targets to Sphinx using the new
+# "make mode" option.  $(O) is meant as a shortcut for $(SPHINXOPTS).
+%: Makefile
+	@$(SPHINXBUILD) -M $@ "$(SOURCEDIR)" "$(BUILDDIR)" $(SPHINXOPTS) $(O)
```
Comment (lucafavatellawork) [inline docs/Makefile]: Could you please confirm this is hooked to CI? I do not see it.

File: sparkl_cli/test/test_vars.py [hunk 4]
```diff
@@ -65,3 +65,6 @@ def test_var_insert(self):
             ])
         baz = result["content"][1]
         assert baz["tag"] == "baz"
+
+    def test_reset(self):
```
Comment (lucafavatellawork) [inline sparkl_cli/test/test_vars.py]: Are you relying on this test being executed at the end of the list of tests of the module hence acting as a teardown? If so I would rather expect this to be rewritten as a teardown - either on test-scope or module-scope.

File: docs/Makefile (hunk 3 above)
Comment (miklosduma) [reply docs/Makefile]: It's not because right now the docs are not built. `usage.rst` is properly compiled on Github and the scripts are run as doctests. At some later stage I'm planning to migrate the existing Github wiki to rst and hook in the build but right now I have to focus on our demo.

File: docs/usage.rst [hunk 5]
```diff
@@ -0,0 +1,233 @@
...
+Using the SPARKL CLI, we can import configurations directly to our user
+tree. For example, `this
+script <https://github.com/opensparkl/examples/blob/master/Examples/importXmp.sh>`__
+imports all our sample configurations to a selected user account.
+
+.. code-block:: python
+    :class: collapsible
+
+    >>> sparkl('ls')
+    {'tag': 'content', 'attr': {'folder': '/', 'count': 1}, 'content': [{'tag': 'folder', 'attr': {..., 'name': 'Scratch'}}]}
```
Comment (lucafavatellawork) [inline docs/usage.rst]: Could you please review all instances of `Scratch`.

**Comment by lucafavatellawork** [comment]
I understand @miklosduma plans some amendments so I am taking the liberty of assigning this PR to him.

File: sparkl_cli/test/test_vars.py (hunk 4 above)
Comment (miklosduma) [reply sparkl_cli/test/test_vars.py]: Added as teardown

File: docs/usage.rst (hunk 5 above)
Comment (miklosduma) [reply docs/usage.rst]: Scratch removed

File: .travis.yml [hunk 6]
```diff
@@ -14,4 +14,4 @@ script:
   - make lint compile
   - CLAIM=postoffice sse start ./sse1
   - sleep 5
-  - make default-test
+  - make default-test
```
Comment (lucafavatellawork) [inline .travis.yml]: This unnecessary change may conflict with #33.

File: docs/usage.rst [hunk 7]
```diff
@@ -0,0 +1,244 @@
...
+    > https://raw.githubusercontent.com/opensparkl/examples/master/Examples/PrimesExpr/Primes_expr.xml \
+    > Test
+
+    change
+    id: ...
+
+The ``sparkl put`` command imports the specified SPARKL configuration - ``Primes_expr.xml`` - to the specified folder - ``Test`` - in your user tree.
+
+You can import the following:
+- A configuration on your file system using the path to it
```
Comment (lucafavatellawork) [inline docs/usage.rst]: This list is rendered unpleasantly by github. Is the bug in github, or in the .rst not complying to the restructuredtext spec? (From a quick look, I did not find anything anomalous in this compared to rst spec.)

Comment (miklosduma) [reply docs/usage.rst]: Probably a github bug, it should read all right

File: .travis.yml (hunk 6 above)
Comment (miklosduma) [reply .travis.yml]: Oops, some git woodo magic, I can see no actual change

File: docs/Makefile [hunk 8]
```diff
@@ -0,0 +1,20 @@
+# Minimal makefile for Sphinx documentation
```
Comment (jacobythwaites) [inline docs/Makefile]: I'd like to see this referenced (even if only in `make help`) in the top-level Makefile so you're led to it without having to know about it.

Can the sphinx dependency be mentioned in help too (e.g. how to install)

File: docs/make.bat (hunk 1 above)
Comment (jacobythwaites) [reply docs/make.bat]: It's a moot point about leaving boilerplate in situ, or cleaning it up.

I think clean it up, we don't support Windoze builds!

File: docs/make.bat [hunk 9]
```diff
@@ -0,0 +1,35 @@
...
+	set SPHINXBUILD=sphinx-build
+)
+set SOURCEDIR=.
+set BUILDDIR=_build
+
+if "%1" == "" goto help
+
+%SPHINXBUILD% >NUL 2>NUL
+if errorlevel 9009 (
+	echo.
```
Comment (jacobythwaites) [inline docs/make.bat]: Ah, I like this (cf earlier comment) but it didn't appear anywhere when I typed `make`

File: docs/usage.rst [hunk 10]
```diff
@@ -0,0 +1,244 @@
+####################
```
Comment (jacobythwaites) [inline docs/usage.rst]: This looks fantastic - but how do I run it?

File: Makefile (hunk 2 above)
Comment (jacobythwaites) [reply Makefile]: Would it be possible to change the name of the `default-test` target to `ci_test` (now I understand @lucafavatellawork 's point about that being the test target run by ci)?

Comment (lucafavatellawork) [reply Makefile]: > Would it be possible to change the name of the `default-test` target to `ci_test` (now I understand @lucafavatellawork 's point about that being the test target run by ci)?

@jacobythwaites  I will prepare separate PR for that.

Comment (lucafavatellawork) [reply Makefile]: @jacobythwaites https://github.com/opensparkl/sse_cli/pull/34

File: docs/Makefile (hunk 8 above)
Comment (miklosduma) [reply docs/Makefile]: Removed sphinx thingy for now, only left the actual doctests in place.

File: docs/make.bat (hunk 9 above)
Comment (miklosduma) [reply docs/make.bat]: Removed sphinx thingy for now, only left the actual doctests in place.

File: docs/usage.rst (hunk 10 above)
Comment (miklosduma) [reply docs/usage.rst]: The Makefile includes running the doctests along the regular tests

**Comment by jacobythwaites** [comment]
Leaving this in situ for the time being, not convinced it's needed.

**Comment by lucafavatellawork** [comment]
@miklosduma Can this be closed following #54?

**Comment by miklosduma** [comment]
Worthwhile content has been reworked and merged via #54 . Dropping this branch
